### PR TITLE
Allow more options in Quantity LaTeX formatting

### PR DESCRIPTION
**Pull request by Juanlu001**
See https://github.com/astropy/astropy/pull/6502 for a first (failed) attempt. This is less aggressive, keeps backwards compatibility everywhere, and prepares the ground for changing the default in the future.

### CODE REVIEW

**Comment by mgeier** [comment]
Please have a look at https://github.com/sympy/sympy/pull/15625.

I think it's a very good idea to get rid of the double dollars for the default behavior.

I would also consider changing the default option to:

```python
                None: ("$\\displaystyle ", "$"),
```

**Comment by Juanlu001** [comment]
@mgeier Just in case, let me clarify some things:

> I think it's a very good idea to get rid of the double dollars for the default behavior.

Actually, `_repr_latex_` outputs single dollars at the moment. This used to be a bit problematic to me (that's why I reported https://github.com/spatialaudio/nbsphinx/issues/119 originally), but although GitHub still does not render `$ ... $` as math, Read the Docs + nbsphinx now does:

* Before http://docs.poliastro.space/en/v0.7.0/examples/Exploring%20the%20New%20Horizons%20launch.html#Option-a):-Insert--from-report,-check--at-parking-perigee
* After http://docs.poliastro.space/en/v0.11.1/examples/Exploring%20the%20New%20Horizons%20launch.html#Option-a):-Insert-C_3-from-report,-check-v_e-at-parking-perigee

This PR was meant to provide a sensible path to make `Quantity._repr_latex_` use double dollar signs in the future, while offering other alternatives for people relying on the single dollar signs version (see https://github.com/astropy/astropy/pull/6502#issuecomment-337719019). However, if the assumption that double dollar signs work better on notebooks is no longer true (not sure what changed in the middle, by the way) we can just remove the note about the future change, keep the new `Quantity.to_string` method (which hopefully is useful on its own) and discuss better defaults.

I see you have worked on this issue already across IPython and SymPy (see https://github.com/jupyter/notebook/issues/4060 and issues/prs linked there). I don't have time to go through all of that right now, but perhaps it's time for the Jupyter core team to provide some guidelines for external packages.

> I would also consider changing the default option to: `None: ("$\\displaystyle ", "$"),`

In my understanding, the default output of `Quantity.to_string` should be as simple as possible and equal to the current `str(quantity)`, with no use of TeX.

**Comment by Juanlu001** [comment]
It was merged but then rolled back (afk, can't look for link)

File: astropy/units/quantity.py [hunk 1]
```diff
@@ -1186,7 +1194,35 @@ def complex_formatter(value):
...
 
-        return r'${0} \; {1}$'.format(latex_value, latex_unit)
+        delimiter_left, delimiter_right = formats[format][subfmt]
+
+        return r'{left}{0} \; {1}{right}'.format(latex_value, latex_unit,
+                                                 left=delimiter_left,
+                                                 right=delimiter_right)
+
+    # Display
+    # TODO: we may want to add a hook for dimensionless quantities?
```
Comment (mhvk) [inline astropy/units/quantity.py]: While you are at it, could you move this up to `_unit_str`?

File: astropy/units/quantity.py [hunk 2]
```diff
@@ -1139,6 +1127,26 @@ def _repr_latex_(self):
...
+        if unit is not None and unit != self.unit:
+            return self.to(unit).to_string(
+                unit=None, precision=precision, format=format, subfmt=subfmt)
+
+        formats = {
+            None: None,
+            "latex": {
+                None: ("$", "$"),
+                "inline": ("$", "$"),
+                "display": (r"$$", r"$$"),
```
Comment (mhvk) [inline astropy/units/quantity.py]: Does a notebook work with `$\displaystyle`? If not, one other suggestion is to call this "$$" or "jupyter". Or actually keep it as `display`, but set it to `$\displaystyle` and worry about `jupyter` in the next PR, when we do `_repr_latex_`.

**Comment by mhvk** [comment]
p.s. This also needs an entry in `CHANGES.txt` under `New Features`.

**Comment by Juanlu001** [comment]
Thanks @mhvk! I addressed your comments. With a bit more time I would add the unicode format as well, but I'd prefer to wait for a future PR.

File: astropy/units/quantity.py [hunk 3]
```diff
@@ -1134,16 +1123,61 @@ def _repr_latex_(self):
         This is treated separately because the numpy default of 1000 is too big
         for most browsers to handle.
 
+        Parameters
+        ----------
+        unit : `~astropy.units.UnitBase`, optional
+            Specifies the unit.  If not provided,
+            the unit used to initialize the quantity will be used.
+
+        precision : , optional
```
Comment (adrn) [inline astropy/units/quantity.py]: Maybe:
```suggestion
        precision : numeric, optional
```

**Comment by Juanlu001** [comment]
Hell yeah GitHub suggestions :guitar:

**Comment by Juanlu001** [comment]
There were two (apparently) unrelated test failures so I rebased && force pushed on current master just in case.

**Comment by Juanlu001** [comment]
Oops, the errors are still there. I'll try to give it a closer look this week.

**Comment by mhvk** [comment]
@Juanlu001 - seems very unlikely the error is due to you. Indeed, I think I've seen that matplotlib error before; cc @bsipocz, @astrofrog (both `mpldev` and `html-docs` fail).

**Comment by pllim** [comment]
Perhaps due to #8347?

**Comment by bsipocz** [comment]
I've definitely seen these errors before.

**Comment by Juanlu001** [comment]
Rebased after https://github.com/astropy/astropy/pull/8352.

**Comment by Juanlu001** [comment]
https://github.com/astropy/astropy/pull/8357 was closed, so we either wait for a scikit-image release or merge this already :)

**Comment by pllim** [comment]
Merging. Thanks for your patience! 😄

**Comment by Juanlu001** [comment]
Thanks! I of course could have waited for scikit-image, but I'm glad we merged :)

**Pull request merged**